storm-client/test/py/test_storm_cli.py

```diff
@@ -67,6 +67,25 @@ def test_jar_command(self):
             ])
         )
 
+        self.mock_execvp.reset_mock()
+
+        self.base_test([
+            'storm', 'jar', '/path/to/jar.jar', 'some.Topology.Class',
+            '-name', 'run-topology', 'randomArgument', '-randomFlag', 'randomFlagValue', '-rotateSize', '0.0001',
+            '-hdfsConf', 'dfs.namenode.kerberos.principal.pattern=hdfs/*.EV..COM'
+        ], self.mock_execvp, mock.call(
+            self.java_cmd, [
+                self.java_cmd, '-client','-Ddaemon.name=', '-Dstorm.options=',
+                '-Dstorm.home=' + self.storm_dir + '', '-Dstorm.log.dir=' + self.storm_dir + "/logs",
+                '-Djava.library.path=', '-Dstorm.conf.file=', '-cp',
+                '' + self.storm_dir + '/*:' + self.storm_dir + '/lib:' + self.storm_dir
+                + '/extlib:/path/to/jar.jar:' + self.storm_dir + '/conf:' + self.storm_dir + '/bin:',
+                '-Dstorm.jar=/path/to/jar.jar', '-Dstorm.dependency.jars=', '-Dstorm.dependency.artifacts={}',
+                'some.Topology.Class', '-name', 'run-topology', 'randomArgument', '-randomFlag', 'randomFlagValue',
+                '-rotateSize', '0.0001', 'dfs.namenode.kerberos.principal.pattern=hdfs/*.EV..COM', '-hdfsConf'
```

Comment:
Why is this correct?
`'dfs.namenode.kerberos.principal.pattern=hdfs/*.EV..COM', '-hdfsConf'`? Should `-hdfsConf` appear before `'dfs.namenode.kerberos.principal.pattern=hdfs/*.EV..COM'`

Reply:
Good catch - the filtered flags are re-added at the end. The argument is of the form -hdfsConf=someValue so the order shouldn't matter